pkg/apis/resource/fuzzer/fuzzer.go

```diff
@@ -18,23 +18,29 @@ package fuzzer
 
 import (
 	fuzz "github.com/google/gofuzz"
-
+	"k8s.io/apimachinery/pkg/runtime"
 	runtimeserializer "k8s.io/apimachinery/pkg/runtime/serializer"
 	"k8s.io/kubernetes/pkg/apis/resource"
 )
 
 // Funcs contains the fuzzer functions for the resource group.
 var Funcs = func(codecs runtimeserializer.CodecFactory) []interface{} {
 	return []interface{}{
-		func(obj *resource.ResourceClaimSpec, c fuzz.Continue) {
-			c.FuzzNoCustom(obj) // fuzz self without calling this function again
+		func(r *resource.DeviceRequest, c fuzz.Continue) {
+			c.FuzzNoCustom(r) // fuzz self without calling this function again
 
-			// Custom fuzzing for allocation mode: pick one valid mode randomly.
-			modes := []resource.AllocationMode{
-				resource.AllocationModeImmediate,
-				resource.AllocationModeWaitForFirstConsumer,
+			// Match defaulter.
+			if r.CountMode == "" {
```

Comment:
To be honest, I am not sure what the "correct" behavior for manual fuzzer functions are. In existing code, I see comments about ["matching the defaulting"](https://github.com/kubernetes/kubernetes/blob/2ad04a0505ef13915e11d8b2b3445e514065a9bb/pkg/apis/core/fuzzer/fuzzer.go#L109-L117), so that is what I am doing also here. I remember that tests were failing when not doing this. Let's try simplifying...

Hmm, nothing obviously fails when removing the fuzzer for `resource.DeviceRequest`. I'll ask on Slack.

Reply:
I think our use of fuzzing here is fuzzy (sorry).  There are times when fuzzing within valid space is useful and there are times when breaking assumptions is useful.  We mostly seem to use fuzzing for round-trip, so if the test passes without special-cases, that seems best, I think.

Reply:
Re-attching in a new comment